source/s3_util.c

```diff
@@ -352,9 +355,13 @@ void aws_s3_add_user_agent_header(struct aws_allocator *allocator, struct aws_ht
 
     const struct aws_byte_cursor space_delimiter = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL(" ");
     const struct aws_byte_cursor forward_slash = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("/");
-
-    const size_t user_agent_product_version_length =
-        g_user_agent_header_product_name.len + forward_slash.len + g_s3_client_version.len;
+    struct aws_byte_cursor platform_cursor = aws_s3_get_current_platform_info()->instance_type;
```

Comment:
Discussion point: This call can result in an imds call for one time. Should we define a new API which returns instance type only if we know it already?

Reply:
I have updated it to only used cached results.